File: new_insertion_sort.swift
#!/usr/bin/env swift
// insertion sort

func insertionSort(list: [Int]) -> [Int] {
  var newList = list
  for i in 1..<list.count {
    let e = newList[i]
    var j = i-1
    while j >= 0 && newList[j] > e {
      newList[j+1] = newList[j]
      j -= 1
    }
    newList[j+1] = e
  }
  return newList
}

let o = [10,8,100,1,637,234,233,235,98,1000,2,5,6]
print(insertionSort(o))

File: new_insertion_sort.js
#!/usr/bin/env node
// insertion sort

function insertionSort(list) {
  var newList = list.slice();
  for (var i = 1; i < newList.length; i++) {
    var value = newList[i];
    var j = i - 1;
    while (j >= 0 && newList[j] > value) {
      newList[j+1] = newList[j];
      j--;
    }
    newList[j+1] = value;
  }
  return newList;
}

var t = [100,2,56,200,-52,3,99,33,177,-199];
console.log(insertionSort(t));
